erts/emulator/beam/erl_bif_persistent.c

```diff
@@ -395,49 +389,101 @@ BIF_RETTYPE persistent_term_put_2(BIF_ALIST_2)
         }
     }
 
-    {
-        Uint term_size;
-        Uint lit_area_size;
-        ErlOffHeap code_off_heap;
-        ErtsLiteralArea* literal_area;
-        erts_shcopy_t info;
-        Eterm* ptr;
+    do_update(ctx);
+    suspend_updater(BIF_P);
+
+    BUMP_REDS(BIF_P, (max_iterations - iterations_until_trap) / ITERATIONS_PER_RED);
+    ERTS_BIF_YIELD_RETURN(BIF_P, am_ok);
+}
+
+BIF_RETTYPE persistent_term_put_new_2(BIF_ALIST_2)
```

Comment:
I think this shares a bit too much with `put_2`, why not merge them both into a separate function with the relevant `Export*` (for trapping) and a flag to control the behavior on key clashes?